CONTRIBUTING.md

```diff
@@ -37,17 +44,51 @@ Do not hesitate to create an issue or pull request directly on the
 The Taipy team manages its backlog in private. Each issue that will be done during our current sprint is
-attached to the `current sprint`. Please, do not work on it, the Taipy team is on it.
+attached to the "current sprint". Please, do not work on it, the Taipy team is on it.
 
-## Code organisation
+All other issues are sorted by labels and free to be taken. If you are new to the project, you can start with the
+"good first issue" or "🆘 Help wanted" label. You can also start with issue with higher priority like "Critical"
+or "High". The higher the priority, the more value it will bring to the project.
+
+If you want to work on an issue, please add a comment and wait to be assigned to the to inform the community that
+you are working on it.
+
+### Contribution workflow
 
-Taipy is organised in five main repositories:
+1. Make your [own fork](https://help.github.com/en/github/getting-started-with-github/fork-a-repo) of the repository
+   target by the issue. Clone it on our local machine, then go inside the directory.
 
-- [taipy-config](https://github.com/Avaiga/taipy-config).
-- [taipy-core](https://github.com/Avaiga/taipy-core).
-- [taipy-gui](https://github.com/Avaiga/taipy-gui).
-- [taipy-rest](https://github.com/Avaiga/taipy-rest).
-- [taipy](https://github.com/Avaiga/taipy) brings previous packages in a single one.
+2. We are working with [Pipenv](https://github.com/pypa/pipenv) for our virtualenv.
+   Create a local env and install development package by running `$ pipenv install --dev`, then run tests with
+   `$ pipenv run pytest` to verify your setup.
+
+3. For convention help, we provide a [pre-commit](https://pre-commit.com/hooks.html) file.
+   This tool will run before each commit and will automatically reformat code or raise warnings and errors based on the
+   code format or Python typing.
+   You can install and setup it up by doing:
+   ```bash
+   $ pipenv install pre-commit
+   $ pipenv run python -m pre-commit install
+   ```
+
+4. Make the changes.<br/>
+   You may want to also add your GitHub login as a new line of the `contributors.txt` file located at the root
+   of this repository. We are using it to list our contributors in the Taipy documentation
+   (see the "Contributing > Contributors" section) and thank them.
+
+5. Create a [pull request from your fork](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork).<br/>
+   Keep your pull request as __draft__ until your work is finished.
+   Do not hesitate to add a comment for help or questions.
+   Before you submit a pull request for review from your forked repo, check that it meets these guidelines:
+     - The code and the branch name follow the [Taipy coding style](#coding-style-and-best-practices).
+     - Include tests.
+     - Code is [rebase](http://stackoverflow.com/a/7244456/1110993).
+     - License is present.
+     - pre-commit works - without mypy error.
+     - GitHub's actions are passing.
```

Comment:
Can they run the GitHub actions on their forked project? I am not certain.

Reply:
On the PR, we need to run the action

Reply:
Yes.
So as a contributor, I can't "check that it meets this guideline" if I cannot run the actions by myself. That is the Taipy maintainer team responsibility. 
Am I missing something here?

Reply:
How about
```suggestion
     - Taipy tests are passing.
```

Reply:
Yep. I like it better. Thx!